docs/adrs/00005-license-export-analytics.md

```diff
@@ -0,0 +1,332 @@
+           .column_as(license::Column::Text, "text")
+           .column_as(license::Column::SpdxLicenses, "spdx_licenses")
+           .column_as(
+              license::Column::SpdxLicenseExceptions,
+              "spdx_license_exceptions",
+           )
+           .into_model::<SbomLicenseBase>()
+           .all(&ctx.db)
+           .await?;
+  
+   Ok(())
+}
+```
+As shown in the above example, when the SPDX SBOM was created, the sbom_name and sbom_namespace could not be obtained. Additionally, we are currently storing both licenseDeclared and licenseConcluded, and we are unable to distinguish between them.
+At same time, if the SBOM is cycloneDX, sbom_name, component_group, component_version will miss.
+
+**The information in the ExtractedLicensingInfos section is completely missing.**
+## Decision
+
+### Current design proposal
+#### Data model
+The current design requires adding a license_type field in purl_license_assertion, as well as a new entity called extracted_licensing_infos. Additionally, the Sbom entity should include fields such as sbom_name, sbom_namespace, component_group, and component_version.
+
+![erd.png](erd.png)
+```rust
+pub enum LicenseCategory {
+    #[sea_orm(string_value = "slc")]
+    SpdxDeclared,
+    #[sea_orm(string_value = "sld")]
+    SpdxConcluded,
+    #[sea_orm(string_value = "clci")]
+    CydxLcid,
+    #[sea_orm(string_value = "clcn")]
+    CydxLcname,
+    #[sea_orm(string_value = "cle")]
+    CydxLexpression,
+    #[sea_orm(string_value = "cd")]
+    ClearlyDefined,
+    #[sea_orm(string_value = "o")]
+    Other,
+}
+```
+
+Currently, the system's licenses come from the following sources: 1) SPDX's declared license and concluded license, 2) CycloneDX, and 3) clearlyDefined. The license export only requires SPDX's declared license and CycloneDX licenses, while concluded licenses and clearlyDefined licenses are not needed. Therefore, the system must distinguish between them.
+
+SpdxDeclared: The license from spdx sbom's declered license.
+SpdxConcluded: The license from spdx sbom's concluded license.
+CydxLcid: The license from CycloneDX sbom's id.
+CydxLcname: The license from CycloneDX sbom's name.
+CydxLexpression: The license from CycloneDX sbom's Expression.
+
+#### SBOM ingestion
+Add the following actions during the ingestion process.
+1. add sbom_namespace, component_group, and component_version into Entity sbom.
+2. add LicenseCategory into Entity purl_license_assertion.
+3. create new Entity extracted_licensing_infos.
+
+#### License Extract API
+This section of the code can be divided into three parts.
+1. Data Collection
+   - This section of the code can refer to [here](#current-license-data-structure-design-in-trustify)    
+
+ 2. Generate the CSV file and compress it. 
+    - This code can refer version one's code.[here](https://github.com/trustification/trustification/blob/main/spog/api/src/license/license_exporter.rs)
+ 3. Endpoint
+    - HTTP GET **/api/v1/sbom/license/{id}**
```

Comment:
This looks like the `id` of the license to me. I assume you mean the `id` of the SBOM?

Reply:
yes sbom id.

Reply:
Then I think it should be swapped around, so that it's clear what the ID refers to: `/api/v2/sbom/{id}/license`.